### PR TITLE
Added new API of NewScalarResourceWithRevocable.

### PR DESCRIPTION
**Pull request by jay-lau**

### CODE REVIEW

File: mesosutil/mesosprotoutil.go [hunk 1]
```diff
@@ -36,6 +36,15 @@ func FilterResources(resources []*mesos.Resource, filter func(*mesos.Resource) b
 	return result
 }
 
+func AddResourceRevocable(resource *mesos.Resource, principal string, role string) *mesos.Resource {
```
Comment (jdef) [inline mesosutil/mesosprotoutil.go]: why are principal and role included as parameters here?

File: mesosutil/mesosprotoutil.go [hunk 2]
```diff
@@ -36,6 +36,15 @@ func FilterResources(resources []*mesos.Resource, filter func(*mesos.Resource) b
...
+        resource.Revocable = &mesos.Resource_RevocableInfo{}
+        return resource
+}
+
+func NewScalarResourceWithRevocable(name string, value float64, principal string, role string) *mesos.Resource {
```
Comment (jdef) [inline mesosutil/mesosprotoutil.go]: I'm not crazy about propagating this anti-pattern of func-name-for-every-resource-type-and-feature. This kind of API nonsense will disappear in mesos:next.

**Comment by jdef** [comment]
thanks for the PR! I'll accept this once AddResourceRevocable is cleaned up

**Comment by jay-lau** [comment]
@jdef It was updated in https://github.com/mesos/mesos-go/pull/227 , my mistake that I generate a new PR

**Pull request closed without merging**